docs/privacy-api.md

```diff
@@ -0,0 +1,259 @@
+			about your browser and how you interact with the website.
+			If you are logged in this information may be associated with
+			your account. You can learn more by visiting the
+			<a href="%s">Google Privacy Policy</a> page.'
+		),
+		'https://policies.google.com/privacy?hl=en'
+	);
+	wp_add_privacy_policy_content( 'google-analytics', $policy_text );
+} );
+```
+
+## Personal Data Exports
+
+When a user makes a request for an export of their personal data a confirmation request should be sent to them via the [ Export Personal Data tool](internal://admin/tools.php?page=export_personal_data) by filling in their email address and clicking send.
+
+![Personal Data Export admin screen](./assets/data-export.png)
+
+Once they have confirmed their request a zip file is created and emailed to them.
+
+### Extending Personal Data Export
+
+By default the export will contain any data known to be associated with the requester's email address such as comments or posts. Some code may extend the platform in such a way that it cannot determine all the data associated with the user automatically.
+
+To extend the data export use the `wp_privacy_personal_data_exporters` filter:
+
+```php
+add_filter( 'wp_privacy_personal_data_exporters', function ( $exporters ) {
+	$exporters['form-responses'] = array(
+		'exporter_friendly_name' => __( 'Form Response Exporter' ),
+		'callback' => 'form_response_exporter',
+	);
+	return $exporters;
+} );
+```
+
+The `form_response_exporter` callback function recieves the requester's email address and is responsible for collecting the custom data.
+
+The function should paginate results to avoid timeouts and performance issues. To help with this core checks the value of `done` in the returned array and if false will call the function again with the `$page` argument incremented by 1.
+
+The following example exports data from a hypothetical `form_responses` custom post type:
+
+```php
+function form_response_exporter( string $email, int $page = 1 ) : array {
+
+	$export_items = [];
+
+	// Fetch form responses from this email address.
+	$responses = new WP_Query( [
+		'posts_per_page' => 300,
+		'paged' => $page,
+		'post_type' => 'form_responses',
+		'meta_key' => 'email',
+		'meta_value' => $meail,
+	] );
+
+	foreach ( $responses->posts as $post ) {
+		// Most item IDs should look like postType-postID
+		// If you don't have a post, comment or other ID to work with,
+		// use a unique value to avoid having this item's export
+		// combined in the final report with other items of the same id.
+		$item_id = "form-responses-{$post->ID}";
+
+		// Core group IDs include 'comments', 'posts', etc.
+		// But you can add your own group IDs as needed.
+		$group_id = 'form-responses';
+
+		// Optional group label. Core provides these for core groups.
+		// If you define your own group, the first exporter to
+		// include a label will be used as the group label in the
+		// final exported report.
+		$group_label = __( 'Form Responses' );
```

Comment:
Maybe add text domains?

Reply:
For the sake of example code it's probably ok but can do